Java/parkservices/src/main/java/com/amazon/randomcutforest/parkservices/threshold/BasicThresholder.java

```diff
@@ -135,11 +138,13 @@ protected double longTermDeviation() {
         return (horizon * primaryDeviation.getDeviation() + (1 - horizon) * secondaryDeviation.getDeviation());
     }
 
-    public double getAnomalyGrade(double score, double factor) {
+    public double getAnomalyGrade(double score, boolean previous, double factor) {
```

Comment:
Yes (last point was anomaly). This test says "if we exceed threshold, we will mark as anomaly. But if we are close to threshold (within the elasticity parameter) and the previous point was potentially an anomaly, then we will pretend as if this score is worth looking into further". What this avoids is when scores go 1.0, 0.99, 1.0, 0.99 and thresholds go 0.99, 1.0, 0.99, 1.0 (note that the threshold only knows up to the previous score). This can happen in all threshold models unless we are careful.

Reply:
got it.  Could you add these explanation as comments?

Reply:
Yes of course. Is everything else ok? If yes then we can keep this PR merged and the add the explanation in the next PR. We should also wait for Josh's comments once he's available.

Reply:
Added more comments.  Please take a look.  I am open to either options.  Whatever is easier for you.